src/test/java/com/team2813/BuildConstantsPublisherTest.java

```diff
@@ -102,4 +103,45 @@ public void publishesBuildConstantsToNetworkTables() {
         .parsesAsLocalDateTime();
     assertThat(getIntegerEntryOrDefault(table, "Dirty", -1)).isAnyOf(0l, 1l);
   }
+
+  @Test
+  public void logsBuildConstantsToConsole() {
+    // Arrange.
+
+    NetworkTableInstance ntInstance = NetworkTableInstance.create();
+    BuildConstantsPublisher publisher = new BuildConstantsPublisher(ntInstance);
+
+    // Keep the original System.out
+    PrintStream originalOut = System.out;
+
+    // Redirect System.out to a ByteArrayOutputStream
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(outputStream));
+
+    // Act.
+    publisher.log();
+
+    // Assert.
+    assertThat(outputStream.toString())
+        .containsMatch(
+            // NOTE that [\r\n]+ is used to match both Windows (\r\n) and Unix (\n) line endings.
+            "MavenName:     Robot2025[\r\n]+"
+                // Matches a Git revision number, e.g., "121"
+                + "GitRevision:   [0-9]+[\r\n]+"
+                // Matches a Git revision hash, e.g., 08205a25fe10c6c6c1ea4db2deabb4aaf4617637
+                + "GitSha:        [0-9a-f]+[\r\n]+"
```

Comment:
Since the commit SHA should always be 40 characters long, you can specify the length
```suggestion
                + "GitSha:        [0-9a-f]{40}[\r\n]+"
```

Reply:
Note: for users without git installed it will be NA, meaning that we may want to use 
`(NA|[0-9a-f]{40})` to allow people without git installed to build code.

Reply:
Hm, wondering if GitRevision will be "NA" as well in this situation?